### ISSUE TITLE
Parser issues

### ISSUE THREAD

**Issue by frafra** [daggaz/json-stream#3]
This  is a valid JSON file: https://api.ohsome.org/v1/elementsFullHistory/geometry?bboxes=9.188295196%2C45.4635324507%2C9.1926242813%2C45.4649771956&properties=metadata&showMetadata=true&time=2007-10-08T00%3A00%3A00Z%2C2021-06-28T09%3A00%3A00Z&types=node

I use `json_stream.requests.load` to load it using requests, with `stream=True`, like this: `data = json_stream.requests.load(resp)`. I try to iterate over `data["features"]`, but the parser fails.

Here is the traceback:

```
Traceback (most recent call last):
  File "/home/frafra/Code/is-osm-uptodate/venv/lib/python3.9/site-packages/werkzeug/wsgi.py", line 462, in __next__
    return self._next()
  File "/home/frafra/Code/is-osm-uptodate/venv/lib/python3.9/site-packages/werkzeug/wrappers/response.py", line 49, in _iter_encoded
    for item in iterable:
  File "/home/frafra/Code/is-osm-uptodate/is-osm-uptodate.py", line 119, in generate
    for feature in data["features"]:
  File "/home/frafra/Code/is-osm-uptodate/venv/lib/python3.9/site-packages/json_stream/base.py", line 35, in _iter_items
    item = self._load_item()
  File "/home/frafra/Code/is-osm-uptodate/venv/lib/python3.9/site-packages/json_stream/base.py", line 168, in _load_item
    item = super()._load_item()
  File "/home/frafra/Code/is-osm-uptodate/venv/lib/python3.9/site-packages/json_stream/base.py", line 128, in _load_item
    raise ValueError(f"Expecting value, comma or ], got {v}")
ValueError: Expecting value, comma or ], got {
```

**Comment by daggaz** [comment]
Ah, yes, the parser falls over if the *first* thing in a list is another list or an object.

This works:
```json
[1, {}]
```

This doesn't
```json
[{}, 1]
```

**Comment by daggaz** [comment]
Fixed in 1.0.2

**Comment by frafra** [comment]
I made a small JSON fuzzer, and I can confirm that the bug has been fixed, and the library seems behaving well:

```
#!/usr/bin/env python3

import json_stream
import json
import io
import random
import string

def random_list(depth):
    items = []
    if depth == 0:
        return items
    for _ in range(random.randrange(10)):
        items.append(random_generic(depth-1))
    return items

def random_dict(depth):
    items = {}
    if depth == 0:
        return items
    for _ in range(random.randrange(10)):
        items[random_item()] = random_generic(depth-1)
    return items

def random_nested(depth):
    return random.choice([random_list, random_dict])(depth)

def random_item(*args):
    return random.choice([
        random.randrange(-100, 100),
        random.random()*100,
        ''.join(random.sample(string.printable, random.randrange(100))),
    ])
    
def random_generic(depth):
    return random.choice([random_item, random_nested])(depth)

def random_json_generator():
    depth = random.randrange(10)
    item = random.choice([random_list, random_dict])(depth)
    return json.dumps(item)

for _ in range(10000):
    random_json = random_json_generator()
    try:
        stream = json_stream.load(io.StringIO(random_json))
        stream.read_all()
    except:
        print(random_json)
```
